src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/X86Base.cs

```diff
@@ -62,5 +67,19 @@ internal abstract class X64
         /// Its functionality is exposed in the public <see cref="System.Numerics.BitOperations" /> class.
         /// </remarks>
         internal static uint BitScanReverse(uint value) => BitScanReverse(value);
+
+        /// <summary>
+        /// void __cpuidex(int cpuInfo[4], int function_id, int subfunction_id);
+        ///   CPUID
+        /// </summary>
+        public static unsafe (int Eax, int Ebx, int Ecx, int Edx) CpuId(int functionId, int subFunctionId)
+        {
+            int* cpuInfo = stackalloc int[4];
+            __cpuidex(cpuInfo, functionId, subFunctionId);
+            return *(((int, int, int, int)*)cpuInfo);
```

Comment:
It's worth calling out `ValueTuple` is `AutoLayout`, so we can't actually use it with `QCall` directly.

This "should" be safe given the current layout algorithm and due to all 4 fields being `float`, but I can always update this to explicitly assign the fields if we think it is a problem.

Reply:
Agree. Though, I think `return (cpuInfo[0], cpuInfo[1], cpuInfo[2], cpuInfo[3]);` would look better.